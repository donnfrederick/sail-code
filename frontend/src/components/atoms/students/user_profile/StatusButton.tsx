import * as React from 'react'

import Button from 'components/atoms/students/Button'
import FavoriteButton from 'components/atoms/students/FavoriteButton'
import ConfirmAddFavorite from 'components/organisms/students/modal_contents/confirm_add_favorite'
import ConfirmRemoveFavorite from 'components/organisms/students/modal_contents/confirm_remove_favorite'
import ConfirmUnblock from 'components/organisms/students/modal_contents/confirm_unblock'

interface Props {
  isBlocked: boolean
  isFavorite: boolean
  userId: number
  open(): void
  setContents(contents: JSX.Element): void
}

export default ({
  isBlocked,
  isFavorite,
  userId,
  setContents,
  open
}: Props) => {
  return isBlocked ? (
    <Button
      type="red"
      text="Blocked"
      fontSize={28}
      width={210}
      height={56}
      onClick={() => {
        setContents(<ConfirmUnblock blockedUserId={userId} />)
        open()
      }}
    />
  ) : (
    <FavoriteButton
      isFavorite={isFavorite}
      onClick={() => {
        setContents(
          isFavorite ? (
            <ConfirmRemoveFavorite favoritedUserId={userId} />
          ) : (
            <ConfirmAddFavorite addFavoritedUserId={userId} />
          )
        )
        open()
      }}
    />
  )
}
