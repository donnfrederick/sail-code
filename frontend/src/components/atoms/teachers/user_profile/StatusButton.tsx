import * as React from 'react'

import Button from 'components/atoms/teachers/Button'
import FavoriteButton from 'components/atoms/teachers/FavoriteButton'
import ConfirmAddFavorite from 'components/organisms/teachers/modal_contents/confirm_add_favorite'
import ConfirmRemoveFavorite from 'components/organisms/teachers/modal_contents/confirm_remove_favorite'
import ConfirmUnblock from 'components/organisms/teachers/modal_contents/confirm_unblock'

interface Props {
  isBlocked: boolean
  isFavorite: boolean
  userId: number
  open(): void
  setContents(contents: JSX.Element): void
}

export default ({ isBlocked, userId, setContents, isFavorite, open }: Props) =>
  isBlocked ? (
    <Button
      type="red"
      text="ブロック中"
      fontSize={28}
      width={210}
      height={56}
      onClick={() => {
        setContents(<ConfirmUnblock blockedUserId={userId} />)
        open()
      }}
    />
  ) : (
    <FavoriteButton
      isFavorite={isFavorite}
      onClick={() => {
        setContents(
          isFavorite ? (
            <ConfirmRemoveFavorite favoritedUserId={userId} />
          ) : (
            <ConfirmAddFavorite addFavoritedUserId={userId} />
          )
        )
        open()
      }}
    />
  )
