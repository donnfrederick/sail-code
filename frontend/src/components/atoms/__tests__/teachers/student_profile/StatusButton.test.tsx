import StatusButton from 'components/atoms/teachers/user_profile/StatusButton'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <StatusButton />', () => {
  const setContentsFn = (contents: JSX.Element) => {}
  const tree = renderer
    .create(
      <StatusButton
        isBlocked={true}
        isFavorite={true}
        userId={1}
        open={() => {}}
        setContents={setContentsFn}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
