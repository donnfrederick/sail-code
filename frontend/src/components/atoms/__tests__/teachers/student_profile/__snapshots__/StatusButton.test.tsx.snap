// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <StatusButton /> 1`] = `
<button
  className="sc-htpNat iyhoTW"
  data-active={true}
  fontSize={28}
  height={56}
  onClick={[Function]}
  width={210}
>
  <div
    className="sc-EHOje jfyosa"
  >
    ブロック中
  </div>
</button>
`;
